### ISSUE TITLE
Configure label name

### ISSUE THREAD

**Issue by efb4f5ff-1298-471a-8973-3d47447115dc** [Levi-Lesches/blocking-issues#4]
It would be nice if i could configure the name of the label that is being added and removed

**Comment by Levi-Lesches** [comment]
I'll try to get to it soon, thanks for the suggestion. If you want to have a go at it yourself in the meantime, I'd be happy to review a PR.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I'm unfortunately not that knowledgeable on GitHub actions but i do know that https://github.com/z0al/dependent-issues has this already implemented maybe you can take a look over there.

**Comment by Levi-Lesches** [comment]
Thanks for your feedback!

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@Levi-Lesches thankyou so much

i have a question about the use of the workflow. Are the following usecases valid for this workflow?

1. PR1 is dependent on PR2 (PR2 has to be merged/closed first in order to get PR1 merged). This one is pretty straight forward.

2. PR1 is dependent on issue1 (issue1 has to be closed first in order to get PR1 merged). This can be helpful when someone already created PR1 but u want issue1 to get addressed first but nobody has created a PR for it yet.

3. issue1 is dependent on PR1 (PR1 has to be merged/closed first in order to close issue1). This can be helpful when somebody forgets to link PR1 as dependency when they create PR2 for issue1

4. issue1 is dependent on issue2 (issue2 has to be closed first in order to close issue1). This can be helpful when there are no PR's yet that addresses one of the issues.

**Comment by Levi-Lesches** [comment]
I've updated the README with a link to https://github.com/Levi-Lesches/blocking-issues-test, my test repo for this bot. Note you can apply branch protection rules to disallow merging if a PR is blocked, which can prevent a blocked PR from being merged. On the other hand, nothing can prevent an issue from being closed since closing means either "not working on" or "already fixed". As to your questions:

1. Yes, https://github.com/Levi-Lesches/blocking-issues-test/pull/26 is blocked by https://github.com/Levi-Lesches/blocking-issues-test/pull/25. The bot marks the former as `blocked` until the latter is closed. 
2. Yes, https://github.com/Levi-Lesches/blocking-issues-test/pull/24 is blocked by https://github.com/Levi-Lesches/blocking-issues-test/issues/22.
3. No, this bot was meant for a workflow where you make issues and then work on them. In that case, there usually _is_ no PR to link to when creating a new issue, and you would simply mark `issue1` as blocked by `issue2` (the issue that `PR1` is fixing). While you _can_ mark an `issue1` as dependent on `PR1`, GitHub does not prevent `PR2` from closing `issue1`, all the bot can do is mark the issue as `blocked`. For accident prevention, you must mark `PR2` as dependent on `PR1` and add branch security so that `PR2` cannot be merged until `PR1` is merged. 
4. Yes, but again, this does not actually prevent anyone from closing the issue. The bot will still comment on the issue and label it as blocked. 

Situations 1, 2, and 4 are the main supported use cases for this workflow (even though scenario 4 does not offer real security, it is still useful to exclude the `blocked` label in searches).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Thankyou so much for ur response. Im going to start testing the workflow on my end to check if everything is running well (ill get back to u if an issue occurs)

**Comment by Levi-Lesches** [comment]
Note that the new feature is in `v1.2`. To not have to manually update your workflow file, you can simply use `v1`, which will always be the most up-to-date version that is still compatible with `v1.1`. 

The new feature is being tracked in #6

**Comment by Levi-Lesches** [comment]
Sorry to add more clutter here; I think I may do #7 instead, which would be a breaking change if you use the new options. With that approach, instead of providing the options to the workflow file, you would make the label yourself, along with its color and description, and then use the `use-label` option to direct the workflow to that label. That makes the workflow more resistant to changes to the label, such as description, color, and name (you would make those changes in the GitHub UI instead).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Test results on

>1. Yes, [This is a PR that's blocked by another PR blocking-issues-test#26](https://github.com/Levi-Lesches/blocking-issues-test/pull/26) is blocked by [This PR is not blocked by anything blocking-issues-test#25](https://github.com/Levi-Lesches/blocking-issues-test/pull/25). The bot marks the former as `blocked` until the latter is closed.

Opened up 5 PR:

1. PR is already merged
2. PR is already closed
3. PR is going to get merged
4. PR is going to get closed 
5. [PR checks if the workflow works correctly by including those PRs as blocked](https://github.com/efb4f5ff-1298-471a-8973-3d47447115dc/FreeTube/pull/423)

On opening up PR 5 the workflow correctly shows me that PR 1 & 2 are not blocked and 3 & 4 do block
After merging PR 3 the workflow doesnt run and the comment from bot stays the same
After closing PR 4 the workflow doesnt run and the comment from the bot stay the same.

will continue testing

**Comment by Levi-Lesches** [comment]
Apologies, since the workflow was designed to have PRs blocked by issues, I left out a few event triggers in the README. I went through the list and updated it (#11), please use the new list there.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
>2. Yes, [A PR that's blocked by an issue.  blocking-issues-test#24](https://github.com/Levi-Lesches/blocking-issues-test/pull/24) is blocked by [Issue 2 blocking-issues-test#22](https://github.com/Levi-Lesches/blocking-issues-test/issues/22).

Tested with 2 issues

1. issue is closed
2. issue will get closed

On opening up the PR it correctly showed me that 1 is closed and 2 is open
On closing issue 2 the workflow triggered correctly

I was just curious what would happen if i would reopen an issue (usecase: closed issue by accident) and it didn't trigger the workflow

**Comment by Levi-Lesches** [comment]
I'm testing the reopening behavior in #8 but hopefully, just adding `reopened` to the triggers list did the trick. I'll update the README once I figure it out or implement something.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@Levi-Lesches Test result on 

>3. No. You _can_  mark an `issue1` as dependent on `PR1`, and the bot will comment and label the issue as `blocked`, but nothing stops `PR2` from saying `fixes issue1` and closing it automatically. For proper protection, mark `PR2` as dependent on `PR1` and add branch security so that `PR2` cannot be merged until `PR1` is merged.

Just did this one out of curiosity. So i created

PR is already merged
PR is already closed
PR is going to get merged
PR is going to get closed
Opened issue thats getting blocked by all 4

on opening issue it displays everything correctly and on closing and merging it doesnt trigger the workflow.

> 4. Yes, but this does not prevent anyone from closing the issue. The bot will still comment on the issue and label it as `blocked`.

Issue blocking issue

linked a closed issue
linked an open issue that is going to be closed
opened issue that is going to be blocked by both

displayed everything correctly on opening. On closing the opened issue workflow didnt trigger.

**Comment by Levi-Lesches** [comment]
Does your workflow file use the new event triggers?

```yaml
on: 
  issues:
    types: [opened, edited, deleted, transferred, closed, reopened]
  pull_request_target: 
    types: [opened, edited, closed, reopened]
```

The important part here is that now, both `issues` and `pull_request_target` have the `closed` trigger ("merged" is `closed` for PRs). Also, make sure that these changes are in your `main` branch, as that is the branch against which actions run.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Yes i use the new event triggers
https://github.com/efb4f5ff-1298-471a-8973-3d47447115dc/FreeTube/blob/workflow/.github/workflows/blocking-issues.yml

I have changed my main branch to be something else to be the default branch. To test this workflow so i dont have to clutter the real main branch.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @Levi-Lesches apologies its been a while. Just wanted to let u know I reran all the tests and all succeeded on v2.

**Comment by Levi-Lesches** [comment]
Thanks! That's very helpful. I still plan to get to the new features too, but I'm glad it's working for now.
